Polish code editor.

jasmussen opened the pull request:
Fixes #21576.

This PR improves the code editor by embracing what it is, a code editor. That means a code font also for the title, and an editor that resizes to the viewport.

Before:

<img width="1021" alt="Screenshot 2020-04-16 at 12 57 52" src="https://user-images.githubusercontent.com/1204802/79448549-e683fc00-7fe1-11ea-9b95-9e87236256d0.png">

After:

![after](https://user-images.githubusercontent.com/1204802/79448556-e8e65600-7fe1-11ea-8ae9-e38bf347e4b9.gif)

It also removes a bunch of dead code.

jasmussen committed:
Polish code editor.

jasmussen committed:
Increase specificity.

mkaz reviewed (approved):
I tested this out and this works well, definite improvement.
No problems on Windows 10 using latest Edge and Firefox 👍 

A minor nit that I'll defer to you, I'd like a little padding between title and content boxes.

jasmussen commented:
Thanks Marcus! 

I actually intentionally removed the spacing between the title and the body, in effort to eliminate borders. Borders pile up and become noise when there's enough of them, so I actualy prefer the simpler look.

shaunandrews commented:
I appreciate the larger space for editing code. Nice work. 

I find the “editing code” indicator and the exit button a little random. They don’t feel intentional and almost look like they’re just floating in space.

Probably outside the scope for this PR, but I’d love to see some basic syntax highlighting. And if we wanna get really fancy, some basic autocomplete/autoclosing for tags.

MichaelArestad commented:
I second everything Shaun said. Nice work!

> I find the “editing code” indicator and the exit button a little random. They don’t feel intentional and almost look like they’re just floating in space.

Are these even necessary?

jasmussen commented:
I was just pointed to #17017 as a good list of aspects to address. For now I wanted to keep things simple and just address low-hanging fruit. But atleast crossposting to that thread is helpful. 

> I find the “editing code” indicator and the exit button a little random. They don’t feel intentional and almost look like they’re just floating in space.

They were designed to align with the editor box:

<img width="1240" alt="Screenshot 2020-04-17 at 18 57 09" src="https://user-images.githubusercontent.com/1204802/79594424-457a6b80-80dd-11ea-82a6-7e512d7aa040.png">

But I see that isn't the case on larger screens. I'll do a little rethink.

> Are these even necessary?

As I recall, we didn't always have them, but we added them exactly because people found it difficult to switch away from this editor once they had activated it. I don't know if @mtias can recall further details, but I distinctly remember this being a thing we added intentionally to solve that problem.

jasmussen commented:
Oh, and CC: @mapk for that bit about whether the indicators were necessary, because I saw that addressed mentioned in the other ticket.

jasmussen committed:
Massage the "editing code" positioning.

jasmussen commented:
> I find the “editing code” indicator and the exit button a little random. They don’t feel intentional and almost look like they’re just floating in space.

I massaged the positioning. Small:

<img width="882" alt="Screenshot 2020-04-21 at 10 57 10" src="https://user-images.githubusercontent.com/1204802/79846644-fbd0a000-83be-11ea-9d1e-fa23140579c9.png">

Medium:

<img width="1236" alt="Screenshot 2020-04-21 at 10 57 15" src="https://user-images.githubusercontent.com/1204802/79846658-fecb9080-83be-11ea-9462-4c6e666dd22e.png">

Large:

<img width="1770" alt="Screenshot 2020-04-21 at 10 57 19" src="https://user-images.githubusercontent.com/1204802/79846667-01c68100-83bf-11ea-997b-1d0a17273325.png">

As noted in [this comment](https://github.com/WordPress/gutenberg/pull/21643#issuecomment-615356747), the label and exit button were added for a reason. The reason can be revisited, of course, but I'd like for that to happen separately.

jasmussen commented:
#17017 outlines a number of ideas for how to improve the code editor. Syntax highlighting, line numbers, a custom sidebar, colors and so on. It also questions the "exit" button. 

What this PR does for now, outside of visual polish and a wider editing interface, is force code fonts for the entire window, making it clearer you're editing code. That addresses part of the issue, but not the whole issue. For that reason, I'd like to merge this as is and keep that ticket open for further improvements.

caban13 commented:
Hi
an editor that resizes to the viewport. when is this coming to WP ? 
I still see a code window of 580 px in WP 5.4.2.
Thanks
Alain

jasmussen commented:
Hello @caban13 

This pull request refreshes the code editor, which resizes to the viewport but has a maximum width of 1032px. This will land in WordPress 5.5.

caban13 commented:
Thank for your answer.
1032 is better than 580, still I don't understand why not 100%, same width as normal view?
Thanks, again
